File: backend/src/modules/version/version.service.ts
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { Version } from './entities/version.entity';

@Injectable()
export class VersionService {
  constructor(
    @InjectRepository(Version)
    private versionRepository: Repository<Version>,
  ) {}

  async findAll(): Promise<Version[]> {
    return this.versionRepository.find({ order: { tableName: 'ASC' } });
  }

  async getVersion(tableName: string): Promise<number | null> {
    const item = await this.versionRepository.findOne({ where: { tableName } });
    return item?.tableVersion ?? null;
  }

  async getStats(): Promise<any> {
    const tables = await this.versionRepository.find();
    return {
      totalTables: tables.length,
      tables: tables.map(t => ({ name: t.tableName, version: t.tableVersion })),
    };
  }
}

File: kam-admin-module-monitor/src/main/java/io/kamailio/admin/modules/version/service/VersionService.java
package io.kamailio.admin.modules.version.service;

import io.kamailio.admin.modules.version.entity.Version;
import io.kamailio.admin.modules.version.repository.VersionRepository;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Map;

@Service
public class VersionService {
    private final VersionRepository repository;

    public VersionService(VersionRepository repository) { this.repository = repository; }

    public List<Version> findAll() {
        return repository.findAll(org.springframework.data.domain.Sort.by("tableName"));
    }

    public Integer getVersion(String tableName) {
        return repository.findByTableName(tableName).map(Version::getTableVersion).orElse(null);
    }

    public Map<String, Object> getStats() {
        var tables = repository.findAll();
        var list = tables.stream().map(t -> Map.<String, Object>of("name", t.getTableName(), "version", t.getTableVersion())).toList();
        return Map.of("totalTables", tables.size(), "tables", list);
    }
}
